src/Type/Compare.php

```diff
@@ -40,10 +41,14 @@ public function process()
             throw new \InvalidArgumentException(sprintf('The operator %s is invalid!', $this->config('operator')));
         }
         foreach ($this->fields() as $field) {
-            $left = $field . ' ' . $this->config('operator');
-            $right = $this->value();
+            $columnType = 'string';
 
-            $conditions[] = [$left => $right];
+            if (is_string($field)) {
+                $columnExists = $this->manager()->table()->schema()->column($field);
+                $columnType = (!$columnExists) ? $this->manager()->table()->schema()->columnType($field) : 'string';
```

Comment:
@Marlinc Oops, sorry. I'm currently creating tests and also added a local variable for this :sweat_smile: